CRM/Contribute/BAO/Contribution.php

```diff
@@ -493,91 +497,45 @@ public static function create(&$params, $ids = []) {
     }
 
     $params['contribution_id'] = $contribution->id;
-    $session = CRM_Core_Session::singleton();
 
     if (!empty($params['note'])) {
       $noteParams = [
         'entity_table' => 'civicrm_contribution',
         'note' => $params['note'],
         'entity_id' => $contribution->id,
-        'contact_id' => $session->get('userID'),
+        'contact_id' => CRM_Core_Session::getLoggedInContactID(),
       ];
       if (!$noteParams['contact_id']) {
         $noteParams['contact_id'] = $params['contact_id'];
       }
       CRM_Core_BAO_Note::add($noteParams);
     }
 
-    // make entry in batch entity batch table
-    if (!empty($params['batch_id'])) {
-      // in some update cases we need to get extra fields - ie an update that doesn't pass in all these params
-      $titleFields = [
-        'contact_id',
-        'total_amount',
-        'currency',
-        'financial_type_id',
-      ];
-      $retrieveRequired = 0;
-      foreach ($titleFields as $titleField) {
-        if (!isset($contribution->$titleField)) {
-          $retrieveRequired = 1;
-          break;
-        }
-      }
-      if ($retrieveRequired == 1) {
-        $contribution->find(TRUE);
-      }
-    }
-
     CRM_Contribute_BAO_ContributionSoft::processSoftContribution($params, $contribution);
 
     $transaction->commit();
 
     $activity = civicrm_api3('Activity', 'get', [
       'source_record_id' => $contribution->id,
       'options' => ['limit' => 1],
-      'sequential' => 1,
       'activity_type_id' => 'Contribution',
       'return' => ['id', 'campaign'],
     ]);
 
     //CRM-18406: Update activity when edit contribution.
-    if ($activity['count']) {
-      // CRM-13237 : if activity record found, update it with campaign id of contribution
-      // @todo compare campaign ids first.
-      CRM_Core_DAO::setFieldValue('CRM_Activity_BAO_Activity', $activity['id'], 'campaign_id', $contribution->campaign_id);
+    if (!empty($activity['id'])) {
       $contribution->activity_id = $activity['id'];
     }
 
-    if (empty($contribution->contact_id)) {
-      $contribution->find(TRUE);
-    }
     CRM_Activity_BAO_Activity::addActivity($contribution, 'Contribution');
 
     // do not add to recent items for import, CRM-4399
     if (empty($params['skipRecentView'])) {
       $url = CRM_Utils_System::url('civicrm/contact/view/contribution',
         "action=view&reset=1&id={$contribution->id}&cid={$contribution->contact_id}&context=home"
       );
-      // in some update cases we need to get extra fields - ie an update that doesn't pass in all these params
-      $titleFields = [
-        'contact_id',
```

Comment:
Ah - so we need these for addRecent. Maybe we just need a function to addContributionRecent that loads extra if need-be